### PR TITLE
Search catalog.toml at multiple default locations

### PR DESCRIPTION
**Pull request by bzwei**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -30,14 +31,20 @@ func main() {
...
+		if configFilePath, err = getConfigFile(); err != nil {
+			panic(err)
+		}
+	}
 	dir, file := filepath.Split(configFilePath)
 	viper.SetConfigName(file)
 	viper.SetConfigType("toml")
 	viper.AddConfigPath(dir)
-	if err := viper.ReadInConfig(); err != nil {
+	if viper.ReadInConfig() != nil {
```
Comment (mkanoor) [inline main.go]: @bzwei Since err has been removed the fmt.Errorf line will be using a bad err

File: main.go [hunk 2]
```diff
@@ -49,6 +56,28 @@ func startRun(config *common.CatalogConfig, rh request.Handler) {
...
 
+func getConfigFile() (string, error) {
+	if fileExists("./rhc/workers/catalog.toml") {
+		return "./rhc/workers/catalog.toml", nil
+	}
+	usr, err := user.Current()
+	if err == nil && fileExists(fmt.Sprintf("%s/.config/rhc/workers/catalog.toml", usr.HomeDir)) {
+		return fmt.Sprintf("%s/.config/rhc/workers/catalog.toml", usr.HomeDir), nil
+	}
+	if fileExists("/etc/rhc/workers/catalog.toml") {
```
Comment (mkanoor) [inline main.go]: @bzwei Should we check if we have the ability to read the file? The file might exist but we won't have access to read it.

Comment (bzwei) [reply main.go]: if the cannot be read, it will blow up when viper read it a few lines below.

File: main.go [hunk 3]
```diff
@@ -49,6 +57,28 @@ func startRun(config *common.CatalogConfig, rh request.Handler) {
 	rh.StartHandlingRequests(config, &towerapiworker.DefaultAPIWorker{})
 }
 
+func getConfigFile() (string, error) {
+	if fileExists("./rhc/workers/catalog.toml") {
+		return "./rhc/workers/catalog.toml", nil
+	}
+	usr, err := user.Current()
+	if err == nil && fileExists(fmt.Sprintf("%s/.config/rhc/workers/catalog.toml", usr.HomeDir)) {
```
Comment (mkanoor) [inline main.go]: @bzwei Should we just a variable to do store the filename once from the Sprintf and then use it in 2 places.

File: main.go [hunk 4]
```diff
@@ -49,6 +57,31 @@ func startRun(config *common.CatalogConfig, rh request.Handler) {
 	rh.StartHandlingRequests(config, &towerapiworker.DefaultAPIWorker{})
 }
 
+func getConfigFile() (string, error) {
+	configFile := "./rhc/workers/catalog.toml"
+	if fileExists(configFile) {
+		return configFile, nil
+	}
+	usr, err := user.Current()
```
Comment (mkanoor) [inline main.go:73]: @bzwei we need to handle this err, if we cannot determine the user for any reason.

Comment (mkanoor) [reply main.go:73]: @bzwei would it make sense to create an array of the filenames and then iterate thru the list till we get the first one
```
func getConfigFile() (string, error) {
filename, _ := range configFiles() {
    if fileExists(filename) {
         return filename, nil
    }
}
    return nil, fmt.Errorf("No config file found")
}

func configFiles() []string {
    var s []string
	s = append(s, "./rhc/workers/catalog.toml")
        usr, err := user.Current()
         if err == nil {
            s= append(s, fmt.Sprintf("%s/.config/rhc/workers/catalog.toml", usr.HomeDir))
        }
	s = append(s, "/etc/rhc/workers/catalog.toml")
	return s
}
```

Comment (bzwei) [reply main.go:73]: My assumption is that the login does not have a user profile, then we should proceed to detect the cofig file under /etc/. If we handle this error and stop here, we'd have no chance to go on.

We could simply log the error but continue with next try. What do you thinik?

Comment (bzwei) [reply main.go:73]: My browser did not auto refresh. I missed you above comment. I will work on it.

**Review by mkanoor — APPROVED** [review]

**Pull request merged**